cmd_cram_demuxlet.cpp

```diff
@@ -199,7 +199,7 @@ int32_t main(int32_t argc, char** argv) {
       continue;
     }
 
-    int32_t n_cleared = vr.clear_buffer_before( bcf_hdr_id2name(vr.cdr.hdr, vr.cursor()->rid), sr.cursor()->core.pos );
+    int32_t n_cleared = vr.clear_buffer_before(  bam_get_chrom(sr.hdr, sr.cursor()), sr.cursor()->core.pos );
```

Comment:
notice the original uses the current variant record for the contig and the current sam-record for the position.  They both should be from the sam-record.